coalib/bearlib/languages/documentation/AssembledDocumentation.py

```diff
@@ -0,0 +1,71 @@
+from coalib.bearlib.languages.documentation.DocumentationComment import (
+    DocumentationComment)
+
+Description = DocumentationComment.Description
+Parameter = DocumentationComment.Parameter
+ReturnValue = DocumentationComment.ReturnValue
+
+
+def assemble_documentation(doccomment, param_sym, ret_sym):
+    """
+    Assembles a list of parsed documentation comments.
+
+    This function just assembles the documentation comment
+    itself, without the markers and indentation.
+
+    :param doccomment:
+        The list of parsed documentation comments.
+    :param param_sym:
+        A two element tuple that defines how a param definition starts and
+        ends in a documentation comment.
+    :param ret_sym:
+        A string that defines how a return statement definition occurs in
+        a documentation comment.
+    """
+    assembled_doc = ""
+    for section in doccomment:
+        section_desc = section.desc.splitlines(keepends=True)
+        if isinstance(section, Description):
+            assembled_doc += section_desc[0]
+        elif isinstance(section, Parameter):
+            assembled_doc += (param_sym[0] + section.name + param_sym[1] +
+                              section_desc[0])
+        elif isinstance(section, ReturnValue):
+            assembled_doc += ret_sym + section_desc[0]
+        for desc in section_desc[1:]:
+            if desc != "":
+                assembled_doc += desc
+    return assembled_doc
+
+
+def assemble_full_documentation(doccomment, param_sym, ret_sym,
+                                markers, indent):
+    """
+    Assembles a list of parsed documentation comments.
+
+    This function assembles the whole documentation comment,
+    with the given markers and indentation.
+
+    :param doccomment:
+        The list of parsed documentation comments.
+    :param param_sym:
+        A two element tuple that defines how a param definition starts and
+        ends in a documentation comment.
+    :param ret_sym:
+        A string that defines how a return statement definition occurs in
+        a documentation comment.
+    :param markers:
+        The markers to be used.
+    :param indent:
+        The indentation level to be used for each line of the documentation
+        comment.
+    """
+    assembled_doc = assemble_documentation(doccomment, param_sym, ret_sym)
+    assembled_doc = assembled_doc.splitlines(keepends=True)
+    assembled = indent + markers[0]
+    for section in assembled_doc:
+        if section == "\n":
+            assembled += '\n'
+        else:
+            assembled += indent + markers[1] + section
+    return assembled + indent + markers[2]
```

Comment:
I think we should make this function an instance method of `DocumentationComment`, while `assemble_documentation` above takes `param_sym`, `ret_sym` and so on and generates a new `DocumentationComment` (like discussed :D).

So it would work like this:
```python
docs = list(extract_documentation(my_file))
doc = docs[0]  # just get some documentation

metadata = doc.metadata
# do whatever you want to do with the metadata

new_doc = assemble_documentation(str(doc), metadata.param, metadata.return)
new_doc.assemble()  # <-- this is the `full_assemble` function
```

You think that makes sense? :D